snuba/query/processors/mapping_optimizer.py

```diff
@@ -264,10 +266,17 @@ def _get_condition_without_redundant_checks(
                 if tag_exist_match:
                     requested_tag = tag_exist_match.string("key")
                     if requested_tag in tag_eq_match_strings:
-                        query.add_experiment("redundant_clause_removed", 1)
-                        continue
+                        if should_apply_optimization:
+                            query.add_experiment("redundant_clause_removed", 1)
+                            # the clause is redundant, thus we continue the loop
+                            # and do not add it to useful_conditions
+                            continue
+                        else:
+                            query.add_experiment("redundant_clause_removed", 0)
                 useful_conditions.append(
-                    self._get_condition_without_redundant_checks(cond, query)
+                    self._get_condition_without_redundant_checks(
+                        cond, query, should_apply_optimization
```

Comment:
Not suggesting this should change now but I think if one more argument were added (especially a bool, Query, or Expression), it might be good to specify the argument names in this call, since the order isn't really semantic in any way.

Reply:
agreed. Hopefully the experiment will be successful and I'll remove this entirely!